controller/replica_controller.go

```diff
@@ -389,6 +390,23 @@ func (rc *ReplicaController) CreatePodSpec(obj interface{}) (*v1.Pod, error) {
 			},
 		}
 	}
+
+	if r.Spec.RestoreName != "" && r.Spec.RestoreFrom != "" {
+		settings, err := rc.ds.GetSetting()
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+		if settings != nil && settings.BackupTargetCredentialSecret != "" {
+			credentialEnv, err := util.ConfigEnvWithCredential(r.Spec.RestoreFrom, settings.BackupTargetCredentialSecret)
```

Comment:
You can just take the container (`pod.Spec.Containers[0]`) as a parameter and make a function to update its environment.